app/helpers/application_helper/button/miq_request.rb

```diff
@@ -1,5 +1,16 @@
 class ApplicationHelper::Button::MiqRequest < ApplicationHelper::Button::GenericFeatureButton
-  needs :@record
+  needs :@record, :@request_tab
+
+  def role_allows_feature?
```

Comment:
I think that we should to have just one implementation of this methofd for all the `MiqRequest*` buttons.

There's a base class `ApplicationHelper::Button::MiqRequest` for these so that one should contain the method.

Furthermore the feature should come from the button's options such as here: https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/application_helper/toolbar/miq_request_center.rb#L39
instead of the `self[:id]` https://github.com/ManageIQ/manageiq-ui-classic/pull/4062/files#diff-aadbc56fce4b8d4f3c4c566f9d009006R12 or a constant: https://github.com/ManageIQ/manageiq-ui-classic/pull/4062/files#diff-7cda4b36d4f0a94e9605b5ac48392c20R12

Also I have noticed some mess between feature names in the buttons. Such as 'approve' vs 'approval' https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/application_helper/toolbar/miq_request_center.rb#L39

It looks really strange, I am not sure how/if it works.

We should probably revisit:
https://github.com/ManageIQ/manageiq-ui-classic/commit/64e85c9927